src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotMatchKeywords.cs

```diff
@@ -1,69 +1,621 @@
+        private static string FormatSymbolName(ISymbol symbol)
+        {
+            return symbol.ToDisplayString(
+                // This format omits the namespace.
+                SymbolDisplayFormat.CSharpShortErrorMessageFormat
+                    // Turn off the EscapeKeywordIdentifiers flag (which is on by default), so that
+                    // a method named "@for" is displayed as "for".
+                    // Turn on the UseSpecialTypes flat (which is off by default), so that parameter
+                    // names of "special" types such as Int32 are displayed as their language alias,
+                    // such as int for C# and Integer for VB.
+                    .WithMiscellaneousOptions(SymbolDisplayMiscellaneousOptions.UseSpecialTypes));
         }
+
+        private ImmutableHashSet<string> s_caseSensitiveKeywords = new string[]
+        {
+            // C#
+            "abstract",
+            "as",
+            "base",
+            "bool",
+            "break",
+            "byte",
+            "case",
+            "catch",
+            "char",
+            "checked",
+            "class",
+            "const",
+            "continue",
+            "decimal",
+            "default",
+            "delegate",
+            "do",
+            "double",
+            "else",
+            "enum",
+            "event",
+            "explicit",
+            "extern",
+            "false",
+            "finally",
+            "fixed",
+            "float",
+            "for",
+            "foreach",
+            "goto",
+            "if",
+            "implicit",
+            "in",
+            "int",
+            "interface",
+            "internal",
+            "is",
+            "lock",
+            "long",
+            "namespace",
+            "new",
+            "null",
+            "object",
+            "operator",
+            "out",
+            "override",
+            "params",
+            "private",
+            "protected",
+            "public",
+            "readonly",
+            "ref",
+            "return",
+            "sbyte",
+            "sealed",
+            "short",
+            "sizeof",
+            "static",
+            "string",
+            "struct",
+            "switch",
+            "this",
+            "throw",
+            "true",
+            "try",
+            "typeof",
+            "uint",
+            "ulong",
+            "unchecked",
+            "unsafe",
+            "ushort",
+            "using",
+            "virtual",
+            "void",
+            "volatile",
+            "while",
+            // Listed as a keywords in Microsoft.CodeAnalysis.CSharp.SyntaxKind, but
+            // omitted, at least for now, for compatibility with FxCop:
+            //"arglist", (Shouldn't this be "__arglist"?)
+            //"makeref",
+            //"reftype",
+            //"refvalue",
+            //"stackalloc",
```

Comment:
This one being absent is more troubling than the other legacy, undocumented, underscore-prefixed keywords above.